### PR TITLE
Create Subject With Task

### PR DESCRIPTION
**Pull request by tanphan313**

### CODE REVIEW

File: app/views/admin/subjects/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  <% end %>
+
+  <%= link_to_add_fields t("subject.create_task"), f, :tasks %>
+
+  <div class="form-group">
+    <div class="col-sm-offset-1 col-sm-11">
+      <%= f.submit t("form.submit"), class: "btn btn-primary" %>
+    </div>
+  </div>
+<% end %>
```
Comment (hienmvvn) [inline app/views/admin/subjects/_form.html.erb]: end file

**Comment by hienmvvn** [comment]
ok

File: app/assets/javascripts/customs.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+$(document).ready(function(){
+  $("form").on("click", ".remove_fields", function(event){
```
Comment (dieunb) [inline app/assets/javascripts/customs.js]: js thì dùng single quote

File: app/models/subject.rb [hunk 3]
```diff
@@ -1,7 +1,20 @@
 class Subject < ActiveRecord::Base
   has_many :course_subjects
   has_many :user_subjects
-  has_many :tasks
+  has_many :tasks, dependent: :destroy
   has_many :courses, through: :course_subjects
   has_many :users, through: :user_subjects
+
+  accepts_nested_attributes_for :tasks, reject_if: ->a{a[:title].blank? || a[:description].blank?},
```
Comment (dieunb) [inline app/models/subject.rb]: convention

File: app/models/subject.rb [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
+    allow_destroy: true
+
+  validates :title, presence: true, length: {minimum: 9, maximum: 90}
+  validates :description, presence: true, length: {minimum: 9}
+  validate :validate
+
+  def validate
+    errors.add :tasks,
```
Comment (dieunb) [inline app/models/subject.rb]: dài quá thì dùng `if` thay vì `unless` nhé

**Comment by hienmvvn** [comment]
ok

File: app/models/subject.rb [hunk 5]
```diff
@@ -1,7 +1,21 @@
...
+
+  accepts_nested_attributes_for :tasks, reject_if:
+    ->task{task[:title].blank? || task[:description].blank?},
+    allow_destroy: true
+
+  validates :title, presence: true, length: {minimum: 9, maximum: 90}
+  validates :description, presence: true, length: {minimum: 9}
+  validate :validate
+
+  def validate
```
Comment (dieunb) [inline app/models/subject.rb]: private function

File: app/models/subject.rb [hunk 6]
```diff
@@ -1,7 +1,21 @@
...
+    if tasks.select{|task| task.present?}.count < 2
```
Comment (dieunb) [inline app/models/subject.rb]: `task.present?` là sao em?

File: app/models/subject.rb (hunk 5 above)
Comment (dieunb) [inline app/models/subject.rb]: private function

File: app/models/subject.rb [hunk 7]
```diff
@@ -1,7 +1,21 @@
 class Subject < ActiveRecord::Base
   has_many :course_subjects
   has_many :user_subjects
-  has_many :tasks
+  has_many :tasks, dependent: :destroy
   has_many :courses, through: :course_subjects
   has_many :users, through: :user_subjects
+
+  accepts_nested_attributes_for :tasks, reject_if:
```
Comment (dieunb) [inline app/models/subject.rb]:
```
 +  accepts_nested_attributes_for :tasks, 
 + reject_if: ->task{task[:title].blank? || task[:description].blank?},
 +    allow_destroy: true
```

**Comment by hienmvvn** [comment]
ok

File: app/models/subject.rb [hunk 8]
```diff
@@ -1,7 +1,20 @@
...
+    reject_if: ->task{task[:title].blank? || task[:description].blank?},
+    allow_destroy: true
+
+  validates :title, presence: true, length: {minimum: 9, maximum: 90}
+  validates :description, presence: true, length: {minimum: 9}
+  validate :validate
+
+  private
+  def validate
+    errors.add :tasks, I18n.t("subject.error.at_least_two_task") unless tasks.count > 1
```
Comment (dieunb) [inline app/models/subject.rb]: too long (< 80 chars)

**Pull request merged**